### ISSUE TITLE
Auto-generated methods / fields for events and properties are not marked as such

### ISSUE THREAD

**Issue by realvictorprm** [dotnet/fsharp#4050]
During using Code Lens I noticed in some cases multiple annotations which was obviously a bug. Cause of such were auto-generated methods / fields by the compiler which we're all on the same line (because they are virtual but need some kind of placement). To fix this I tried filtering each `FSharpMemberOfFunctionOrValue` element out which `func.IsCompilerGenerated` was true. However I noticed that several methods were not flagged although they are compiler generated. 

Example:

```fsharp
type Foo() =

    let foo = new Event<int>()

    member __.bar1 =
        foo.Publish

    [<CLIEvent>] 
    member __.bar2 =
        foo.Publish  
```

The compiler generates here getter and setter for both members but does not flag them as compiler generated. Also the generated methods are not flagged as event neither as getter or setter method for events (e.g. `IsEventAddMethod`or `IsEventRemoveMethod`) which I think is wrong too.

#### Repro Steps

1. Create a new F# project
2. Add the F# Compiler Service package to it
3. Iterate over func declarations of the example code below (use the checker)
4. Print out whether methods are compiler generated

#### Expected behavior
Methods which are compiler generated should have the flag `IsCompilerGenerated` set to true.
Moreover relevant flags like `IsEventAddMethod` should be set to true too for events.

Here a text file which contains the output for the remove method of `bar2` from the example:
[EventAutogeneratedBug.txt](https://github.com/Microsoft/visualfsharp/files/1523841/EventAutogeneratedBug.txt)

**Comment by TIHan** [comment]
If there are serious issues with some not being marked as compiler generated, then we should try to fix them. Perhaps would could be smarter on what is compiler generated.

**Comment by realvictorprm** [comment]
this issue is still present and should be fixed before #3608 is merged.

The code lens don't break because of this but you'll get duplicate data.

**Comment by realvictorprm** [comment]
Last weekend I tried fixing it again without much success. My investigation returned that there isn't set anything for event properties identifying them as such. I can open a PR which adds a test covering the bug so we can discuss what should and shouldn't be.

And again, this badly needs to be fixed for code lens. If not fixed we're going to see 3 annotations for a single event member.
